### DISCUSSION TITLE
Support Style Props for Svelte

### DISCUSSION THREAD

**Discussion by devunt** [chakra-ui/panda#2287] [Ideas]
### Description

Currently Panda CSS only supports style props for jsx-based components.
This request proposes style props support for svelte components.

### Problem Statement/Justification

`Component.svelte`
```svelte
<script lang="ts">
  import { css } from '$styled-system/css';
  import type { SystemStyleObject } from '$styled-system/types';

  let cssProp: SystemStyleObject | undefined = undefined;
  export { cssProp as css };
</script>

<div class={css(cssProp)} />
```

`+page.svelte`
```svelte
<script lang="ts">
  import Component from './Component.svelte';
</script>

<Component css={{ color: 'red' }} />
```

This code renders following markup, but no corresponding styles are generated.
```html
<div class="text_red" />
```

Panda generator does not see this as a token references, in contrast to what it does to jsx components.

### Proposed Solution or API

Please support svelte component props like mentioned above.

### Alternatives

Use `css.raw` as it forcefully make Panda generator to recognize it as a tokens.
But it is somewhat tedious to use `css.raw` everywhere.

`+page.svelte`
```svelte
<script lang="ts">
  import { css } from '$styled-system/css';
  import Component from './Component.svelte';
</script>

<Component css={css.raw({ color: 'red' })} />
```

**Comment by astahmer** [comment]
[ts-morph](https://github.com/dsherret/ts-morph) only supports ts(x) syntax
Panda supports svelte (and vue) [by converting svelte DSL to a ts-friendly syntax](https://github.com/chakra-ui/panda/blob/09ebaf2ee4e9eed05874ab8ee4d31bd76b4a64e1/packages/parser/src/svelte-to-tsx.ts#L9) so that ts-morph can parse it and generate valid AST nodes

from a quick look at the existing test, it seems like the `{#each` or `{#if` etc kind of blocks are messing with the AST
so, currently `<Component css={{ color: 'red.100' }} />` isn't parsed correctly but wrapping that inside a `<div>` seems to fix the issue `<div><Component css={{ color: 'red.100' }} /></div>`
minimal repro [here](https://ts-ast-viewer.com/#code/MYewdgzgLgBATgUzAEwXGBeGAeZBLANwD5sALARhmABsBDCCDAb2gE9qEBfIgVwjzABzKnQYwARgPxDsAegpEAUDBWrVipgGJaAd1p5YABzggAtnghdFi7IZH1GTYAwAUTKiGog4ALhgByKARaagA6ADYABkj-GE4ASm49AwFhADNvGChSBBhjMwtcqBB4BAhPAgRQ6rlDJSYfbKQYAhCeKxs6gBUclrbcixgmVup2zlr6n2BaKGBSGDQTOE5rW3sGZmcINw8vXwCgkNCAdmjYhO4AZTMEbNSYHSRYHRMhPyZF71DTMohaQS4Ew0sl0+igKxsAGEzIZwE8qBsmO5QHs-P5EMhQuQznFODBZEpcIQSNDTLCwPCtsxkZ5vGiMVicZw8QS5PhiNYbLJ2UQgA), without a wrapping ts-morph only sees `JsxText` 
<img width="941" alt="image" src="https://github.com/chakra-ui/panda/assets/47224540/924711eb-1360-4450-8987-c881d29b2476"> but with a `<div>` the AST looks good again (it's properly nested in JsxAttribute > Identifier etc)
<img width="993" alt="image" src="https://github.com/chakra-ui/panda/assets/47224540/f32e1bea-0ac7-4a48-a814-3a9c61a51bd0">



feel free to make a PR to fix this for better support of style props in this `svelteToTsx` function

---

an alternative is to use [hooks](https://panda-css.com/docs/concepts/hooks) (`parser:before` and `parser:after`) to make your own parser and extract svelte style props however you want by using svelte tooling
